# CAPN10snp

## Uncovering Genetic Variations in the CAPN10 Gene: A Comprehensive SNP Analysis

### Introduction:
The CAPN10 gene, linked to susceptibility to type 2 diabetes mellitus (T2DM), has received significant attention for its involvement in regulating glucose metabolism. Here, we present an ongoing computational analysis focused on identifying potentially functional single nucleotide polymorphisms (SNPs) within CAPN10. Our study aims to distinguish harmful non-synonymous SNPs (nsSNPs) from neutral variants, utilizing various sequence and structure-based bioinformatic tools. Through a multimodal approach, we meticulously explore a range of nsSNPs, assessing their potential to impact protein structure and function. This comprehensive analysis marks a foundational step towards understanding the functional consequences of genetic variation in CAPN10, with implications for future studies in T2DM and cancer research, including personalized medicine initiatives.

### Aim
This project aims to conduct a comprehensive SNP analysis of the CAPN10 gene to identify potential functional variants associated with T2DM.

### General Objectives:
Retrieve SNP data: Collect high-quality SNP data within the CAPN10 gene region from public genomic databases such as dbSNP and 1000 Genomes Project.

Functional annotation: Predict the functional significance of identified SNPs using bioinformatics tools such as SIFT and PolyPhen-2 to assess their potential impact on protein structure and function.

Linkage disequilibrium analysis: Investigate the patterns of linkage disequilibrium and identify haplotype blocks within the CAPN10 gene region to understand the genetic architecture.

Association analysis: Assess the association of identified SNPs with T2DM susceptibility using available genetic datasets and perform statistical analyses to determine their significance.

Functional validation: Validate the functional relevance of prioritized SNPs using in vitro assays or functional genomics approaches to elucidate their role in T2DM pathogenesis.


### Methodology:
![image](https://github.com/omicscodeathon/capn10snp/blob/main/figures/Workflow_CAPN10snp.jpg)

Figure 1. Workflow

Data retrieval: SNP data within the CAPN10 gene region were retrieved from NCBI and dbSNP. Protein sequences from these genes were obtained and used for the second stage of analysis.

Functional annotation: The functional impact of the identified SNPs were predicted using a series of bioinformatics tools for a more comprehensive analysis. 

![image](figures/SIFT_analysis.jpg)

Figure 2. SIFT analysis


![image](figures/align_GVGD.png)

Figure 3. Align GVGD analysis


![image](figures/panther_analysis.png)

Figure 4. PANTHER analysis

![image]()
Figure 5. Free Energy Stability


![image](figures/HOPE_Mutations.png)

Figure 6. HOPE analysis showing structural changes in the amino acid sequences

![image]()
Figure 5. Protein-Protein Interaction



Linkage disequilibrium analysis: Analyze linkage disequilibrium patterns and identify haplotype blocks using statistical methods.

Association analysis: Conduct association analysis using available genetic datasets and perform statistical tests.

Functional validation: Validate the functional relevance of prioritized SNPs using experimental assays or functional genomics approaches.


### Catalog of identified SNPs within the CAPN10 gene.
Assessment of functional significance for prioritized SNPs.

Identification of haplotype blocks and linkage disequilibrium patterns within CAPN10.

Association analysis results linking specific SNPs to various disorders.

Functional validation of selected SNPs to elucidate their role in T2DM pathogenesis.

### Expected Outcomes:
1.Identification of non-synonymous SNPs (nsSNPs), exploring the possibility that they may alter the structure and function of proteins.
2.Identification of Asssociated disease with human CAPN10 gene.

### Team Members: 
1. Shivani Pawar
2. Nigel Dolling
3. Musa Muhammad Shamsuddeen
4. Gershom Olajire
5. Emmanuel Baluku


